libControls/ToolTip.h

```diff
@@ -9,6 +9,7 @@
 
 #include <utility>
 #include <vector>
+#include <functional>

```

Comment:
Moving the new function declaration to an unnamed namespace also means you can move this include out of the header file. Leaving it in the header means it would be transitively included by anything including the `ToolTip.h` header.

Or, if you want to get rid of this include entirely, you can transform the method into a template, with a template parameter for the function object rather than using `std::function`. If you're not concerned about restricting the template type parameter to ensure it's something useful, the syntax should be fairly simple:
```cpp
template <typename Callback>
void processLines(const std::string& text, const Callback& lineProcessor) const;
```

Of course if you try to use a non-function type for the type parameter, such as an `int`, you may end up with cryptic error messages. There might be a way to restrict the type to a callable, though that would be more work.